packages/shared/src/utils/helpers/duration.ts

```diff
@@ -1,28 +1,77 @@
 import { localize } from '@deriv/translations';
 import { toMoment } from '../date';
 
+type TContract = {
+    max_contract_duration: string;
+    min_contract_duration: string;
+    expiry_type: string;
+    start_type: string;
+};
+
+type TMaxMin = {
+    min: number;
+    max: number;
+};
+
+type TUnit = {
+    text: string;
+    value: string;
+};
+
+type TDurations = {
+    // min_max: {
+    //     spot?: Partial<Record<'tick' | 'intraday' | 'daily', TMaxMin>>;
+    //     forward?: Record<'intraday', TMaxMin>;
+    // };
+    // units_display: Partial<Record<'spot' | 'forward', TUnit[]>>;
+
+    min_max: {
+        // spot: {
+        //     tick: TMaxMin;
+        //     intraday: TMaxMin;
+        //     daily: TMaxMin;
+        // };
+        // forward: {
+        //     intraday: TMaxMin;
+        // };
+    };
+    units_display: {
+        // spot: TUnit[];
+        // forward: TUnit[];
+    };
+};
+
 const getDurationMaps = () => ({
-    t: { display: localize('Ticks'), order: 1 },
+    t: { display: localize('Ticks'), order: 1, to_second: 0 },
     s: { display: localize('Seconds'), order: 2, to_second: 1 },
     m: { display: localize('Minutes'), order: 3, to_second: 60 },
     h: { display: localize('Hours'), order: 4, to_second: 60 * 60 },
     d: { display: localize('Days'), order: 5, to_second: 60 * 60 * 24 },
 });
 
-export const buildDurationConfig = (contract, durations = { min_max: {}, units_display: {} }) => {
-    durations.min_max[contract.start_type] = durations.min_max[contract.start_type] || {};
-    durations.units_display[contract.start_type] = durations.units_display[contract.start_type] || [];
+export const buildDurationConfig = (
+    contract: TContract,
+    durations: TDurations = { min_max: {}, units_display: {} }
+) => {
+    type TDurationMinMax = keyof typeof durations.min_max;
+    type TDurationUnits = keyof typeof durations.units_display;
+    type TDurationMap = keyof typeof duration_maps;
+
+    durations.min_max[contract.start_type as TDurationMinMax] =
+        durations.min_max[contract.start_type as TDurationMinMax] || {};
+    durations.units_display[contract.start_type as TDurationUnits] =
+        durations.units_display[contract.start_type as TDurationUnits] || [];
 
     const obj_min = getDurationFromString(contract.min_contract_duration);
     const obj_max = getDurationFromString(contract.max_contract_duration);
 
-    durations.min_max[contract.start_type][contract.expiry_type] = {
+    durations.min_max[contract.start_type as TDurationMinMax][contract.expiry_type] = {
         min: convertDurationUnit(obj_min.duration, obj_min.unit, 's'),
         max: convertDurationUnit(obj_max.duration, obj_max.unit, 's'),
     };
 
-    const arr_units = [];
-    durations.units_display[contract.start_type].forEach(obj => {
+    const arr_units: [] = [];
```

Comment:
Not sure what is the types of **arr_units**

Reply:
```suggestion
    const arr_units: string[] = [];
```